lib/compat/wordpress-6.0/blocks.php

```diff
@@ -77,16 +77,18 @@ function gutenberg_build_query_vars_from_query_block( $block, $page ) {
 		if ( ! empty( $block->context['query']['categoryIds'] ) || ! empty( $block->context['query']['tagIds'] ) ) {
 			$tax_query = array();
 			if ( ! empty( $block->context['query']['categoryIds'] ) ) {
-				$tax_query[] = array(
+				$category_ids = array_filter( array_map( 'intval', $block->context['query']['categoryIds'] ) );
+				$tax_query[]  = array(
 					'taxonomy'         => 'category',
-					'terms'            => $block->context['query']['categoryIds'],
+					'terms'            => $category_ids,
 					'include_children' => false,
 				);
 			}
 			if ( ! empty( $block->context['query']['tagIds'] ) ) {
+				$tag_ids     = array_filter( array_map( 'intval', $block->context['query']['tagIds'] ) );
```

Comment:
```suggestion
```

Suggestion applies throughout: basically to ensure the data is safe closer to the point it used.

Reply:
Done, thanks Peter!